src/Compilers/CSharp/csc/Program.cs

```diff
@@ -1,19 +1,30 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.IO;
 using Microsoft.CodeAnalysis.CommandLine;
-using Roslyn.Utilities;
-using System;
 
 namespace Microsoft.CodeAnalysis.CSharp.CommandLine
 {
     public class Program
     {
         public static int Main(string[] args)
-            => Main(args, Array.Empty<string>());
+        {
+            try
+            {
+                return MainCore(args);
+            }
+            catch (FileNotFoundException e)
+            {
+                // Catch exception from missing compiler assembly.
+                // Report the exception message and terminate the process.
+                Console.Error.WriteLine(e.Message);
+                return CommonCompiler.Failed;
+            }
+        }
 
-        public static int Main(string[] args, string[] extraArgs)
```

Comment:
Was `public static int Main(string[] args, string[] extraArgs)` part of the public API?